pycbc/workflow/core.py

```diff
@@ -764,6 +764,139 @@ def _filename(self, ifo, description, extension, segment):
         return "%s-%s-%s-%s.%s" % (ifo, description.upper(), start, 
                                    duration, extension)  
     
+class NewFile(File):
+    '''
+    This class holds the details of an individual output file 
+    This file(s) may be pre-supplied, generated from within the workflow
+    command line script, or generated within the workflow. The important stuff
+    is:
+
+    * The ifo that the File is valid for
+    * The time span that the OutFile is valid for
+    * A short description of what the file is
+    * The extension that the file should have
+    * The url where the file should be located
+
+    An example of initiating this class:
+    
+    c = File("H1", "INSPIRAL_S6LOWMASS", segments.segment(815901601, 815902001), file_url="file://localhost/home/spxiwh/H1-INSPIRAL_S6LOWMASS-815901601-400.xml.gz" )
+
+    another where the file url is generated from the inputs:
+
+    c = File("H1", "INSPIRAL_S6LOWMASS", segments.segment(815901601, 815902001), directory="/home/spxiwh", extension="xml.gz" )
+    '''
+
+    def __init__(self, ifos, exe_name, segs, lfn,
+                 extension=None, directory=None, tags=None, 
+                 store_file=True, use_tmp_subdirs=False):
+        """
```

Comment:
Larne, I'm a bit confused at how you resolved my original comments. It now seems that the class *does not* take an lfn (although it calls it that), but in practice is taking a pfn, from which it derives an lfn internally. It in fact will not work if one just provides an lfn (just a filename, no path, url, etc). 

In practice, I think this still covers the case we use commonly, but it is definitely mislabeled then. I think it might make sense to be able to directly create a file object form a pfn, but it should have a properly named object, and should be separate from the lfn. 

So maybe these are the ways to make a file object
1) metadata + extension (filename chosen for you), no pfn automatically, often used for workflow products
2) lfn, where the executable has not control of the output naming scheme, no pfn automatically
3) pfn, derive lfn from pfn, add pfn automatically, used for files that come from outside the workflow